### PR TITLE
Update GNN sampling post processing functions to optionally consider seed vertices in renumbering

### PR DESCRIPTION
**Pull request by seunghwak**
Upon cugraph-pyg requests, this PR updates GNM sampling post processing functions to optionally consider seed vertices in renumbering.

Previously, seed vertices with 0 neighbors (to sample from) are ignored in renumbering. Once this PR is merged and if seed vertices is provided, seed vertices (regardless of whether they appear in the sampled edge list or not) will have a smaller vertex ID than vertices that first appear as seed vertices' neighbors or later hops.

Marked as non-breaking assuming that this functions are only internally used.

### CODE REVIEW

**Comment by alexbarghi-nv** [comment]
I'm tentatively approving this, will give final approval once I test

**Review by ChuckHastings — APPROVED** [review]

**Comment by ChuckHastings** [comment]
Should be able to merge once @alexbarghi-nv approves.  Let us know how testing goes.

**Comment by alexbarghi-nv** [comment]
I'm seeing a bug where memory addresses are getting corrupted.  Not sure if it's on my end or if it's something this PR introduced.  Still troubleshooting.  Will post a MRE if I am able to isolate it to something in this PR.

**Comment by seunghwak** [comment]
> I'm seeing a bug where memory addresses are getting corrupted. Not sure if it's on my end or if it's something this PR introduced. Still troubleshooting. Will post a MRE if I am able to isolate it to something in this PR.

Have you set do_expensive_check to true?

**Comment by alexbarghi-nv** [comment]
> > I'm seeing a bug where memory addresses are getting corrupted. Not sure if it's on my end or if it's something this PR introduced. Still troubleshooting. Will post a MRE if I am able to isolate it to something in this PR.
> 
> Have you set do_expensive_check to true?

I'll try that

**Comment by alexbarghi-nv** [comment]
It looks like it may be something I introduced in my PR

**Comment by alexbarghi-nv** [comment]
I couldn't replicate this with only the code in this PR

**Comment by alexbarghi-nv** [comment]
> I'm seeing a bug where memory addresses are getting corrupted. Not sure if it's on my end or if it's something this PR introduced. Still troubleshooting. Will post a MRE if I am able to isolate it to something in this PR.

I have now isolated and fixed the bug.  It was an issue with data being improperly copied in my PR.

File: cpp/src/c_api/uniform_neighbor_sampling.cpp [hunk 1]
```diff
@@ -296,19 +296,21 @@ struct uniform_neighbor_sampling_functor : public cugraph::c_api::abstract_funct
...
+              std::move(wgt),
+              std::move(edge_id),
+              std::move(edge_type),
+              std::move(hop),
+              std::nullopt,
+              std::nullopt,
+              offsets ? std::make_optional(
+                          raft::device_span<size_t const>{offsets->data(), offsets->size()})
                       : std::nullopt,
+              edge_label ? edge_label->size() : size_t{1},
```
Comment (alexbarghi-nv) [inline cpp/src/c_api/uniform_neighbor_sampling.cpp:312]:
```suggestion
              (label_list_ == nullptr) ? size_t{1} : label_list_->size_,
```

File: cpp/src/c_api/uniform_neighbor_sampling.cpp [hunk 2]
```diff
@@ -327,21 +329,21 @@ struct uniform_neighbor_sampling_functor : public cugraph::c_api::abstract_funct
...
+                                         std::move(dst),
+                                         std::move(wgt),
+                                         std::move(edge_id),
+                                         std::move(edge_type),
+                                         std::move(hop),
+                                         offsets
+                                           ? std::make_optional(raft::device_span<size_t const>{
+                                               offsets->data(), offsets->size()})
+                                           : std::nullopt,
+                                         edge_label ? edge_label->size() : size_t{1},
```
Comment (alexbarghi-nv) [inline cpp/src/c_api/uniform_neighbor_sampling.cpp:343]:
```suggestion
                                         (label_list_ == nullptr) ? size_t{1} : label_list_->size_,
```

**Review by alexbarghi-nv — CHANGES_REQUESTED** [review]
Small change request.  Otherwise good.

File: cpp/src/c_api/uniform_neighbor_sampling.cpp (hunk 1 above)
Comment (alexbarghi-nv) [reply cpp/src/c_api/uniform_neighbor_sampling.cpp:312]: never mind, I think this is ok as is

File: cpp/src/c_api/uniform_neighbor_sampling.cpp (hunk 2 above)
Comment (alexbarghi-nv) [reply cpp/src/c_api/uniform_neighbor_sampling.cpp:343]: same here, resolving

**Review by alexbarghi-nv — APPROVED** [review]
Approved, but we can't integrate this with dask+Python until that code gets refactored (#4358)

This is ready to go for the new distributed sampling framework, though, which just uses `pylibcugraph` and NCCL.

👍

**Pull request merged**